src/bio.c

```diff
@@ -322,3 +406,34 @@ void bioKillThreads(void) {
         }
     }
 }
+
+void bioPipeReadJobCompList(aeEventLoop *el, int fd, void *privdata, int mask) {
+    UNUSED(el);
+    UNUSED(mask);
+    UNUSED(privdata);
+
+    char buf[128];
+    list *tmp_list = NULL;
+
+    while (read(fd, buf, sizeof(buf)) == sizeof(buf));
+
+    /* Handle event loop events if pipe was written from event loop API */
+    pthread_mutex_lock(&bio_mutex_comp);
+    if (listLength(bio_comp_list)) {
+        tmp_list = bio_comp_list;
+        bio_comp_list = listCreate();
```

Comment:
```suggestion
        bio_comp_list = listCreate();
        listSetFreeMethod(bio_comp_list, zfree);
```

Reply:
then we can remove zfree() in the following whiile loop

Reply:
Unless you have strong opinion about it, I prefer not to do this change because list is created at two different places which requires to add it at both.